### ISSUE TITLE
[BUG] Expected to find the value declaration of symbol

### ISSUE THREAD

**Issue by Disane87** [compodoc/compodoc#585]
##### **Overview of the issue**

After installation and generating the doc with the command, according to your deocumentation, we get the following error. We cannot identify what causes this problem

```Unhandled Rejection at: Promise {
  <rejected> Error: Expected to find the value declaration of symbol 'LoginComponent'.
    at InvalidOperationError.BaseError [as constructor] (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\BaseError.js:7:28)
    at new InvalidOperationError (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\InvalidOperationError.js:8:28)
    at Object.throwIfNullOrUndefined (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\helpers.js:96:15)
    at Symbol.getValueDeclarationOrThrow (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\compiler\common\Symbol.js:66:23)
    at RouterParserUtil.cleanFileDynamics (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:1375:18)
    at Dependencies.getSourceFileDecorators (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5812:45)
    at E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5638:27
    at Array.map (<anonymous>)
    at Dependencies.getDependencies (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5632:21)
    at CliApplication.Application.getDependenciesData (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:7309:40) } reason: Error: Expected to find the value declaration of symbol 'LoginComponent'.
    at InvalidOperationError.BaseError [as constructor] (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\BaseError.js:7:28)
    at new InvalidOperationError (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\InvalidOperationError.js:8:28)
    at Object.throwIfNullOrUndefined (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\errors\helpers.js:96:15)
    at Symbol.getValueDeclarationOrThrow (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\ts-simple-ast\dist\compiler\common\Symbol.js:66:23)
    at RouterParserUtil.cleanFileDynamics (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:1375:18)
    at Dependencies.getSourceFileDecorators (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5812:45)
    at E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5638:27
    at Array.map (<anonymous>)
    at Dependencies.getDependencies (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:5632:21)
    at CliApplication.Application.getDependenciesData (E:\Projekte\VS2017\Project\AngularFrontend\ClientApp\node_modules\@compodoc\compodoc\dist\index-cli.js:7309:40)
```

login.component.ts

```TypeScript
import { Component, OnInit } from '@angular/core';
import { Router, ActivatedRoute } from '@angular/router';
import swal, { SweetAlertResult } from 'sweetalert2';
import { AuthenticationService } from '../_services/authentication.service';
import Sweetalert2 from 'sweetalert2';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { debug } from 'util';
import { UserService } from '../_services/user.service';
import { TranslateService } from '@ngx-translate/core';

@Component({
    moduleId: module.id,
    styleUrls: ['login.component.scss'],
    templateUrl: 'login.component.html'
})

export class LoginComponent implements OnInit {
    model: any = {};
    loading = false;
    returnUrl: string;

    private authWindow: Window;
    failed: boolean;
    error: string;
    errorDescription: string;
    isRequesting: boolean;
    public externalProvider: ExternalProvider;

    myStyle: object = {};
    myParams: object = {};
    width: number = 100;
    height: number = 100;

    constructor(
        private http: HttpClient,
        private route: ActivatedRoute,
        private router: Router,
        private authenticationService: AuthenticationService,
        private userService: UserService,
        private translateService: TranslateService) {
        //this.externalProvider.useGoogleAuth = false;
        //this.externalProvider.useMicrosoftAuth = false;
        //this.externalProvider.useWindowsLogin = false;
        //this.http = myhttp;

        this.http.get<ExternalProvider>('/ExternalProvider').subscribe(result => {
            this.externalProvider = result;
        }, error => console.error(error));
    }

    ngOnInit() {
        // reset login status
        //this.authenticationService.logout();

        // get return url from route parameters or default to '/'
        let returl = this.route.snapshot.queryParams['returnUrl'] || '/';
        if (returl != '') {
            this.returnUrl = returl;
        }
    }
    login() {
        this.loading = true;
        this.authenticationService.login(this.model.username, this.model.password)
            .subscribe(
            data => {
                this.router.navigate([this.returnUrl]);
            },
            response => {
                //this.alertService.error(error);
                this.loading = false;

                switch (response.status) {
                    case 401:
                        swal(this.translateService.instant("ERROR"), this.translateService.instant("LOGIN_ERROR_USERNAME_OR_PW_INCORRECT"), "error");
                        break;

                    default:
                        swal(response.statusText, response.error.message, "error");
                }

            });
    }

    //externalLogin(provider: string) {
    //  console.log("External Login: " + provider);
    //  var externalLoginModel: ExternalLoginModel = new ExternalLoginModel();
    //  externalLoginModel.Provider = provider;
    //  externalLoginModel.ReturnUrl = this.returnUrl;

    //  let headers: HttpHeaders = new HttpHeaders({ 'Content-Type': 'application/json; charset=utf-8' });//, "Access-Control":"Allow - Origin"
    //  //this.http.post("/auth/ExternalLogin", externalLoginModel, { headers }).subscribe(result => {
    //  //  console.log(result.toString()); /*window.open(result.toString());*/
    //  //});
    //  this.authWindow = window.open('https://accounts.google.com/o/oauth2/auth?response_type=code&client_id=549694845659-sdipfgbficscu3stgidjkgmdhpoedp1d.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A54401%2FAuth%2FExternalLoginCallback&scope=openid%20profile%20email&state=CfDJ8CpTbbKMZPpKj8OiGboaBpfcIDACMGyDVJVk0qizKe9Rtv6-xRCuTdG5-UQeTyPxrtVKtjweZql1U8Ak9_wRfZJIFiYDcYDswY1Rv5qUhaBpMrm7ahb9gfr_kk3azaaanhGC7X4_8u1LfxeDK3vm9MTnjbeR85RzA30TU3mjVUbT_qt8XM_FNTIngYf7oKB2PeDd89-OJ5iOShw7_-WHrjZ_xFCDE7hL_G80Zqe1N9RD',null,'width=600,height=400');
    //  }

    externalLogin(provider: string) {
        switch (provider) {
            case 'google':
                this.authenticationService.googleLogin();
                break;
            case 'microsoft':
                this.authenticationService.microsoftLogin(this.externalProvider);
                break;
            case 'windows':
                this.authenticationService.windowsLogin();
                break;
        }

    }

}


export interface ExternalProvider {
    useWindowsLogin: boolean;   // = false;
    useGoogleAuth: boolean;     // = false;
    useMicrosoftAuth: boolean;  // = false;

    microsoftAuthUrl: string;
    microsoftAuthID: string;
    microsoftAuthParams: string;
    microsoftAuthRedirectUrl: string;
    microsoftAuthSize: string;
}

class ExternalLoginModel {
    Provider: string;
    ReturnUrl: string;
}
``` 

If we comment out our app.module.routes.ts then the doc is generated. 
app.module.routes,ts
```TypeScript

export const routes: Routes = [
    { path: 'login', component: components.LoginComponent, data: { title: 'Login' } },
    {
        path: '', canActivateChild: [AuthGuard],
        children: [
            { path: 'home', component: components.HomeComponent, data: { title: 'Home' } },
            { path: 'counter', component: components.CounterComponent, data: { title: 'Counter' } },
            { path: 'fetch-data', component: components.FetchDataComponent, data: { title: 'Fetch-Data' } },
            { path: 'chat', component: components.ChatComponent, data: { title: 'Chat' } },
            { path: 'timeandexpenses', component: components.TimeAndExpensesComponent, data: { title: 'timeandexpenses' } },
            { path: 'timeandexpenses/:employee', component: components.TimeAndExpensesComponent, data: { title: 'timeandexpenses' } },
            { path: 'timeandexpenses/:employee/:date', component: components.TimeAndExpensesComponent, data: { title: 'timeandexpenses' } },
            { path: 'report', component: components.ReportComponent, data: { title: 'Report' } },

            { path: '**', redirectTo: 'home', data: { title: 'Home' } }
        ]
    }
];
```

##### **Operating System, Node.js, npm, compodoc version(s)**

Operation System: Windows 10 x64 Enterprise
Node: v8.11.2
Compodoc: "@compodoc/compodoc": "^1.1.3",

##### **Angular configuration, a `package.json` file in the root folder**
```
{
    "name": "AngularFrontend",
    "version": "0.0.1",
    "license": "MIT",
    "scripts": {
        "ng": "ng",
        "start": "ng serve --extract-css",
        "build": "ng build --extract-css",
        "build:ssr": "npm run build -- --app=ssr --output-hashing=media",
        "test": "ng test",
        "lint": "ng lint",
        "e2e": "ng e2e",
        "compodoc": "./node_modules/.bin/compodoc -p src/tsconfig.app.json"
    },
    "private": true,
    "dependencies": {
        "@angular/animations": "^5.2.11",
        "@angular/common": "^5.2.11",
        "@angular/compiler": "^5.2.11",
        "@angular/core": "^5.2.11",
        "@angular/forms": "^5.2.11",
        "@angular/http": "^5.2.11",
        "@angular/platform-browser": "^5.2.11",
        "@angular/platform-browser-dynamic": "^5.2.11",
        "@angular/platform-server": "^5.2.11",
        "@angular/router": "^5.2.11",
        "@aspnet/signalr": "^1.0.0-rc1-update1",
        "@fortawesome/angular-fontawesome": "0.1.0-9",
        "@fortawesome/fontawesome": "^1.1.8",
        "@fortawesome/fontawesome-svg-core": "^1.2.0-14",
        "@fortawesome/free-brands-svg-icons": "^5.1.0-8",
        "@fortawesome/free-regular-svg-icons": "^5.1.0-11",
        "@fortawesome/free-solid-svg-icons": "^5.1.0-11",
        "@nguniversal/module-map-ngfactory-loader": "^5.0.0",
        "@ngx-translate/core": "^9.1.1",
        "@ngx-translate/http-loader": "^3.0.1",
        "@syncfusion/ej2-ng-dropdowns": "^16.1.46",
        "@syncfusion/ej2-ng-grids": "^16.1.44",
        "angular-particle": "^1.0.4",
        "angular2-ladda": "^2.0.0",
        "angular2-notifications": "^1.0.2",
        "aspnet-prerendering": "^3.0.1",
        "bootstrap": "^4.1.1",
        "core-js": "^2.5.7",
        "devextreme": "^18.1.3",
        "devextreme-angular": "^18.1.3",
        "jquery": "3.3.1",
        "ng-select": "^1.0.0-rc.5",
        "ngx-bootstrap": "^2.0.5",
        "ngx-loading": "^1.0.14",
        "ngx-moment": "^2.0.0-rc.0",
        "ngx-responsive": "^5.0.8",
        "popper.js": "1.14.3",
        "rxjs": "^5.5.11",
        "simple-line-icons": "^2.4.1",
        "sweetalert2": "^7.21.1",
        "zone.js": "^0.8.26"
    },
    "devDependencies": {
        "@angular/cli": "~1.7.4",
        "@angular/compiler-cli": "^5.2.11",
        "@angular/language-service": "^5.2.11",
        "@compodoc/compodoc": "^1.1.3",
        "@types/jasmine": "^2.8.7",
        "@types/jasminewd2": "~2.0.2",
        "@types/node": "^10.1.3",
        "codelyzer": "^4.3.0",
        "jasmine-core": "^3.1.0",
        "jasmine-spec-reporter": "~4.2.1",
        "karma": "^2.0.2",
        "karma-chrome-launcher": "~2.2.0",
        "karma-coverage-istanbul-reporter": "^2.0.1",
        "karma-jasmine": "^1.1.2",
        "karma-jasmine-html-reporter": "^1.1.0",
        "protractor": "^5.3.2",
        "ts-node": "~4.1.0",
        "tslint": "^5.10.0",
        "typescript": "^2.6.2"
    }
}
```

##### **Compodoc installed globally or locally ?**

locally in project

##### **Motivation for or Use Case**

Documentation is not generated

**Comment by vogloblinsky** [comment]
in your `app.module.routes,ts` file, what is `components` ?

**Comment by Disane87** [comment]
This is out comoponent alias:
```ts
import * as components from './app.module.components'
```
It's only a collection of all available view components:

```ts
// Generelles Layout der Seite
export { FullLayoutComponent, LAYOUT_COMPONENTS, LAYOUT_DIRECTIVES } from './_layout';

// Seiten
export { HomeComponent } from './home/home.component';
export { CounterComponent } from './counter/counter.component';
export { FetchDataComponent } from './fetch-data/fetch-data.component';
export { ChatComponent } from './chat/chat.component';
export { LoginComponent } from './login/login.component';
export { TimeAndExpensesComponent } from './timeandexpenses/timeandexpenses.component';
export { ReportComponent } from './report/report.component';
export { SearchComponent } from './search/search.component';
export { MockupComponent } from './mockup/mockup.component';

```

**Comment by vogloblinsky** [comment]
Ok thanks

**Comment by Disane87** [comment]
Ana updates about this?

**Comment by luisfilipecvs** [comment]
Hi, sorry for repeating Disane87 question but this issue is blocking my project from updating to latest version of compodoc. 

In my case the problem comes from having other properties from the route object assign by an object. 
eg:
{
                path: 'somePath',
                data: {
                    features: [GRANT_FEATURES.A_GRANT_PERMISSION]
                },
                canLoad: [RoutePermissionGuard],
                loadChildren: '../@someModule'
},

**Comment by luisfilipecvs** [comment]
I would like to add more info regarding this:

the error is 
[20:17:49] Analysing routes definitions and clean them if necessary
Unhandled Rejection at: Promise {
  <rejected> Error: Expected to find the value declaration of symbol 'A_GRANT_PERMISSION'.

In the case of the example I gave in the comment above GRANT_FEATURES is a Object.freeze({A_GRANT_PERMISSION: 'someString'})

**Comment by luisfilipecvs** [comment]
pleeeeaaase can you give some kind of feedback regarding this?

**Comment by Disane87** [comment]
Any updates about this?

**Comment by CalogeroMandracchia** [comment]
Hi! Me and @nicodemoraffaele also have the same bug! I tried downgrading till the "1.1.0" but no luck :/

**Comment by rafasoyyo** [comment]
No updates ???

**Comment by Disane87** [comment]
Really? Won‘t fix?

**Comment by Disane87** [comment]
???

**Comment by luisfilipecvs** [comment]
This issue as been here long time. I'm blocked on an older version and can't updated because of this.

**Comment by DiegoAlvesGarcia** [comment]
Por algum motivo quando o servidor do compodoc está subindo ele não entende o que é, no seu caso, o import 'components ' ( import * as components from './app.module.components' ).
Se você importar seus componentes diretamente no .routes.ts funcionará.

@vogloblinsky, por que isso ocorre? Poderia verificar, por favor?

**Comment by AdditionAddict** [comment]
Wow this bot is annoying - hard to know how to register I've encountered same issue / whether to open new issue / if project is supported still? In order words if its worth my time producing minimal example

Related to barrel imports? Seems to not complain on other modules and then fails. No work arounds above.

```
[07:41:53] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/checkout/checkout.module.ts
[07:41:53] found          : CheckoutModule
[07:41:53]                : - imports:
[07:41:53]                :     - CommonModule
[07:41:53]                :     - ReactiveFormsModule
[07:41:53]                :     - RouterModule
[07:41:53]                :     - MaterialModule
[07:41:53]                :     - SharedModule
[07:41:53]                : - declarations:
[07:41:53]                :     - ...fromContainers.containers
[07:41:53]                :     - ...fromComponents.components
[07:41:53]                :     - CheckoutReviewPdfComponent
[07:41:53]                : - providers:
[07:41:53]                :     - SuppliersGuard
[07:41:53]                :     - DeliveryLocationsGuard
[07:41:53]                :     - EmployeesGuard
[07:41:53] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/dashboard/dashboard-routing.module.ts
[07:41:53] Analysing routes definitions and clean them if necessary
[07:42:02] Routes parsing error, maybe a trailing comma or an external variable, trying to fix that later after sources scanning.
[07:42:02] found          : DashboardRoutingModule
[07:42:02]                : - imports:
[07:42:02]                :     - DashboardModule
[07:42:02]                :     - RouterModule
[07:42:02]                : - exports:
[07:42:02]                :     - RouterModule
[07:42:02] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/dashboard/dashboard.module.ts
[07:42:02] found          : DashboardModule
[07:42:02]                : - imports:
[07:42:02]                :     - CommonModule
[07:42:02]                :     - RouterModule
[07:42:02]                :     - MaterialModule
[07:42:02]                : - declarations:
[07:42:02]                :     - ...fromContainers.containers
[07:42:02]                :     - ...fromComponents.components
[07:42:02]                : - providers:
[07:42:02]                :     - ReportService
[07:42:02] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/delivery/delivery-routing.module.ts
[07:42:03] Analysing routes definitions and clean them if necessary
[07:42:12] Routes parsing error, maybe a trailing comma or an external variable, trying to fix that later after sources scanning.
[07:42:12] found          : DeliveryRoutingModule
[07:42:12]                : - imports:
[07:42:12]                :     - DeliveryModule
[07:42:12]                :     - RouterModule
[07:42:12]                : - exports:
[07:42:12]                :     - RouterModule
[07:42:12] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/delivery/delivery.module.ts
[07:42:12] found          : DeliveryModule
[07:42:12]                : - imports:
[07:42:12]                :     - CommonModule
[07:42:12]                :     - RouterModule
[07:42:12]                :     - ReactiveFormsModule
[07:42:12]                :     - MaterialModule
[07:42:12]                :     - SharedModule
[07:42:12]                : - declarations:
[07:42:12]                :     - ...fromContainers.containers
[07:42:12]                :     - ...fromComponents.components
[07:42:12] parsing        : C:/Users/andrewa/Desktop/ipad-app/projects/procurement-app/src/app/delivery-location/delivery-location-routing.module.ts
[07:42:12] Analysing routes definitions and clean them if necessary
Unhandled Rejection at: Promise {
  <rejected> Error: Expected to find the value declaration of symbol 'DeliveryLocationsComponent'.
      at InvalidOperationError.BaseError [as constructor] (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\BaseError.js:7:28)
      at new InvalidOperationError (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\InvalidOperationError.js:8:28)
      at Object.throwIfNullOrUndefined (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\helpers.js:96:15)
      at Symbol.getValueDeclarationOrThrow (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\compiler\common\Symbol.js:72:23)
      at RouterParserUtil.cleanFileDynamics (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5669:18)
      at AngularDependencies.getSourceFileDecorators (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8152:46)
      at C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
      at Array.map (<anonymous>)
      at AngularDependencies.getDependencies (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7949:21)
      at CliApplication.Application.getDependenciesData (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:9498:40) } reason: Error: Expected to find the value declaration of symbol 'DeliveryLocationsComponent'.
    at InvalidOperationError.BaseError [as constructor] (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\BaseError.js:7:28)
    at new InvalidOperationError (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\InvalidOperationError.js:8:28)
    at Object.throwIfNullOrUndefined (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\errors\helpers.js:96:15)
    at Symbol.getValueDeclarationOrThrow (C:\Users\andrewa\Desktop\ipad-app\node_modules\ts-simple-ast\dist\compiler\common\Symbol.js:72:23)
    at RouterParserUtil.cleanFileDynamics (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5669:18)
    at AngularDependencies.getSourceFileDecorators (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8152:46)
    at C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
    at Array.map (<anonymous>)
    at AngularDependencies.getDependencies (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7949:21)
    at CliApplication.Application.getDependenciesData (C:\Users\andrewa\Desktop\ipad-app\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:9498:40)
```
